cvat/apps/quality_control/quality_reports.py

```diff
@@ -1970,18 +1970,24 @@ def _find_closest_unmatched_shape(shape: dm.Annotation):
     # row/column index in the confusion matrix corresponding to unmatched annotations
     _UNMATCHED_IDX = -1
 
-    def _make_zero_confusion_matrix(self) -> Tuple[List[str], np.ndarray]:
-        label_names = [
-            label.name
-            for label in self._gt_dataset.categories()[dm.AnnotationType.label]
-            if not label.parent
-        ]
+    def _make_zero_confusion_matrix(self) -> Tuple[List[str], np.ndarray, Dict[int, int]]:
+        label_id_idx_map = {}
+        label_names = []
+        max_label_id = -1
+        for label_id, label in enumerate(self._gt_dataset.categories()[dm.AnnotationType.label]):
+            if not label.parent:
+                label_id_idx_map[label_id] = len(label_names)
+                label_names.append(label.name)
+
+            max_label_id = label_id
+
+        label_id_idx_map[max_label_id + 1] = len(label_names)
```

Comment:
Why is this needed? It shouldn't be possible for the client(s) of this function to index the map with this key.

Reply:
This is needed to make the matrix size and label names list the same size, mostly to avoid confusion.

Reply:
It's still confusing, because `max_label_id + 1` can be a valid label ID (if there's a sublabel following the last non-sublabel).  IMO, this map entry is unnecessary and you shouldn't add it.